src/reflect/scala/reflect/internal/Constants.scala

```diff
@@ -233,23 +233,43 @@ trait Constants extends api.Constants {
       else if (tag == ClazzTag) signature(typeValue)
       else value.toString()
 
-    def escapedChar(ch: Char): String = (ch: @switch) match {
-      case '\b' => "\\b"
-      case '\t' => "\\t"
-      case '\n' => "\\n"
-      case '\f' => "\\f"
-      case '\r' => "\\r"
-      case '"'  => "\\\""
-      case '\'' => "\\\'"
-      case '\\' => "\\\\"
-      case _    => if (ch.isControl) "\\u%04X".format(ch.toInt) else String.valueOf(ch)
-    }
-
     def escapedStringValue: String = {
-      def escape(text: String): String = text flatMap escapedChar
+      import java.lang.StringBuilder
+      def escapedChar(b: StringBuilder, ch: Char): Boolean = {
+        def quadNibble(x: Int, i: Int): Unit =
+          if (i < 4) {
+            quadNibble(x >> 4, i + 1)
+            val n = x & 0xF
+            val c = if (n < 10) '0' + n else 'A' + (n - 10)
+            b.append(c.toChar)
+          }
+        val replace = (ch: @switch) match {
+          case '\b' => "\\b"
+          case '\t' => "\\t"
+          case '\n' => "\\n"
+          case '\f' => "\\f"
+          case '\r' => "\\r"
+          case '"'  => "\\\""
+          case '\'' => "\\\'"
+          case '\\' => "\\\\"
+          case _ if ch.isControl => b.append("\\u"); quadNibble(ch.toInt, 0); return true;
+          case _    => b.append(ch); return false
+        }
+        b.append(replace)
+        true
+      }
+      def escape(text: String): String = {
+        val max = text.length
+        val b = new StringBuilder(max).append("\"")
+        def loop(i: Int, modified: Boolean): Boolean =
+          if (i < max) loop(i + 1, escapedChar(b, text.charAt(i)) || modified)
+          else modified
+        val _ = loop(i = 0, modified = false) // could discard buffer and let intrinsic "\"" + text + "\""
+        b.append("\"").toString
```

Comment:
Looks like this evolved to no longer need/use the `modified`, let's remove it.

Reply:
I was about to, but I hoped it was needed.